diff --git tb/base/android/android_image_reader_compat.cc tb_diff/base/android/android_image_reader_compat.cc
--- tb/base/android/android_image_reader_compat.cc
+++ tb_diff/base/android/android_image_reader_compat.cc
@@ -34,7 +34,11 @@ bool AndroidImageReader::IsSupported() {
 }
 
 AndroidImageReader::AndroidImageReader() {
+#if defined(ENABLE_TRIPLE_BANANA) && defined(ARCH_CPU_ARM64)
+  is_supported_ = false;
+#else
   is_supported_ = LoadFunctions();
+#endif
 }
 
 bool AndroidImageReader::LoadFunctions() {
