diff --git tb/base/feature_list.h tb_diff/base/feature_list.h
--- tb/base/feature_list.h
+++ tb_diff/base/feature_list.h
@@ -231,6 +231,10 @@ class BASE_EXPORT FeatureList {
   // struct, which is checked in builds with DCHECKs enabled.
   static bool IsEnabled(const Feature& feature);
 
+#if defined(ENABLE_TRIPLE_BANANA)
+  static void SetEnabled(const Feature& feature, bool value);
+#endif
+
   // Returns the field trial associated with the given |feature|. Must only be
   // called after the singleton instance has been registered via SetInstance().
   static FieldTrial* GetFieldTrial(const Feature& feature);
