diff --git tb/build/config/compiler/compiler.gni tb_diff/build/config/compiler/compiler.gni
--- tb/build/config/compiler/compiler.gni
+++ tb_diff/build/config/compiler/compiler.gni
@@ -235,7 +235,7 @@ if (strip_absolute_paths_from_debug_symbols_default) {
 # If it wasn't manually set, set to an appropriate default.
 assert(symbol_level >= -1 && symbol_level <= 2, "Invalid symbol_level")
 if (symbol_level == -1) {
-  if (is_android && !is_component_build && !use_thin_lto) {
+  if (is_android && !is_component_build) {
     # Reduce symbol level when it will cause invalid elf files to be created
     # (due to file size). https://crbug.com/648948.
     symbol_level = 1
@@ -288,11 +288,9 @@ if (forbid_non_component_debug_builds) {
 
 # Assert that the configuration isn't going to hit https://crbug.com/648948.
 # An exception is made when target_os == "chromeos" as we only use the Android
-# toolchain there to build relatively small binaries. There is a workaround in
-# place for builds using ThinLTO (through '-generate-type-units' flag), so
-# these are exempt from limitation.
+# toolchain there to build relatively small binaries.
 assert(ignore_elf32_limitations || !is_android || target_os == "chromeos" ||
-           is_component_build || use_thin_lto || symbol_level < 2,
+           is_component_build || symbol_level < 2,
        "Android 32-bit non-component builds cannot have symbol_level=2 " +
            "due to 4GiB file size limit, see https://crbug.com/648948. " +
            "If you really want to try this out, " +
