diff --git tb/base/android/android_image_reader_compat.cc tb_diff/base/android/android_image_reader_compat.cc
--- tb/base/android/android_image_reader_compat.cc
+++ tb_diff/base/android/android_image_reader_compat.cc
@@ -38,6 +38,15 @@ AndroidImageReader::AndroidImageReader() {
 }
 
 bool AndroidImageReader::LoadFunctions() {
+#if defined(ENABLE_TRIPLE_BANANA)
+#if defined(ARCH_CPU_ARM64)
+  // AImageReader is not working on ARM64.
+  if (base::android::BuildInfo::GetInstance()->sdk_int() >=
+      base::android::SDK_VERSION_P) {
+    return false;
+  }
+#endif
+#endif
   // If the Chromium build requires __ANDROID_API__ >= 26 at some
   // point in the future, we could directly use the global functions instead of
   // dynamic loading. However, since this would be incompatible with pre-Oreo
