diff --git tb/base/android/java/src/org/chromium/base/library_loader/LibraryLoader.java tb_diff/base/android/java/src/org/chromium/base/library_loader/LibraryLoader.java
--- tb/base/android/java/src/org/chromium/base/library_loader/LibraryLoader.java
+++ tb_diff/base/android/java/src/org/chromium/base/library_loader/LibraryLoader.java
@@ -344,7 +344,7 @@ public class LibraryLoader {
     // both be used as the basis to ship on L, and the default APK used by developers on 10+.
     private boolean forceSystemLinker() {
         boolean result =
-                mUseChromiumLinker && !mUseModernLinker && Build.VERSION.SDK_INT >= VERSION_CODES.Q;
+                mUseChromiumLinker && !mUseModernLinker && Build.VERSION.SDK_INT >= VERSION_CODES.N;
         if (result) {
             Log.d(TAG,
                     "Forcing system linker, relocations will not be shared. "
