diff --git tb/build/android/gyp/proguard.py tb_diff/build/android/gyp/proguard.py
--- tb/build/android/gyp/proguard.py
+++ tb_diff/build/android/gyp/proguard.py
@@ -239,7 +239,6 @@ def _OptimizeWithR8(options,
 
     cmd = build_utils.JavaCmd(options.warnings_as_errors) + [
         '-Dcom.android.tools.r8.allowTestProguardOptions=1',
-        '-Dcom.android.tools.r8.verticalClassMerging=1',
     ]
     if options.disable_outlining:
       cmd += ['-Dcom.android.tools.r8.disableOutlining=1']
@@ -430,6 +429,14 @@ def _CheckForMissingSymbols(r8_path,
         'Ljavax/management/ObjectInstance',
         'Ljavax/management/ObjectName',
         'Ljavax/management/StandardMBean',
+
+        # For Triple Banana
+        'Landroidx/biometric/BiometricFragment',
+        'Landroidx/biometric/FingerprintDialogFragment',
+        'Landroidx/biometric/FingerprintHelperFragment',
+        'Landroidx/fragment/app/Fragment',
+        'Landroidx/fragment/app/FragmentActivity',
+        'Lorg/chromium/android_webview/common/PlatformServiceBridgeImpl',
     ]
 
     had_unfiltered_items = '  ' in stderr
