diff --git tb/chrome/android/BUILD.gn tb_diff/chrome/android/BUILD.gn
--- tb/chrome/android/BUILD.gn
+++ tb_diff/chrome/android/BUILD.gn
@@ -28,6 +28,7 @@ import("//third_party/icu/config.gni")
 import("//third_party/protobuf/proto_library.gni")
 import("//tools/resources/generate_resource_allowlist.gni")
 import("//tools/v8_context_snapshot/v8_context_snapshot.gni")
+import("//triple_banana/triple_banana.gni")
 import("//weblayer/variables.gni")
 import("channel.gni")
 import("java_sources.gni")
@@ -533,6 +534,7 @@ android_library("chrome_java") {
     "//third_party/gif_player:gif_player_java",
     "//third_party/google_android_play_core:com_google_android_play_core_java",
     "//third_party/metrics_proto:metrics_proto_java",
+    "//triple_banana/modules:mojom_java",
     "//ui/android:ui_java",
     "//ui/base/mojom:mojom_java",
     "//ui/gfx/geometry/mojom:mojom_java",
@@ -575,6 +577,8 @@ android_library("chrome_java") {
   # From java_sources.gni.
   sources = chrome_java_sources + [ app_hooks_impl ]
 
+  sources += triple_banana_hooks_sources + triple_banana_jni_sources
+
   # Include sources from feed_java_sources.gni.
   sources += feed_java_sources
   srcjar_deps += feed_srcjar_deps
@@ -3484,6 +3488,8 @@ generate_jni("chrome_jni_headers") {
     "java/src/org/chromium/chrome/browser/webauth/Fido2Helper.java",
   ]
 
+  sources += triple_banana_jni_sources
+
   # Used for testing only, should not be shipped to end users.
   if (enable_offline_pages_harness) {
     sources += [ "java/src/org/chromium/chrome/browser/offlinepages/evaluation/OfflinePageEvaluationBridge.java" ]
