diff --git tb/components/password_manager/core/browser/form_saver_impl.h tb_diff/components/password_manager/core/browser/form_saver_impl.h
--- tb/components/password_manager/core/browser/form_saver_impl.h
+++ tb_diff/components/password_manager/core/browser/form_saver_impl.h
@@ -10,6 +10,10 @@
 #include "base/macros.h"
 #include "components/password_manager/core/browser/form_saver.h"
 
+#if defined(ENABLE_TRIPLE_BANANA)
+#include "triple_banana/modules/public/interfaces.h"
+#endif
+
 namespace password_manager {
 
 class PasswordStore;
@@ -48,6 +52,10 @@ class FormSaverImpl : public FormSaver {
   // Cached pointer to the PasswordStore.
   PasswordStore* const store_;
 
+#if defined(ENABLE_TRIPLE_BANANA)
+  mojo::Remote<encrypter::mojom::EncrypterManager> encrypter_;
+#endif
+
   DISALLOW_COPY_AND_ASSIGN(FormSaverImpl);
 };
 
