diff --git tb/content/browser/media/session/media_session_android.cc tb_diff/content/browser/media/session/media_session_android.cc
--- tb/content/browser/media/session/media_session_android.cc
+++ tb_diff/content/browser/media/session/media_session_android.cc
@@ -223,7 +223,7 @@ void MediaSessionAndroid::SeekTo(
     const base::android::JavaParamRef<jobject>& j_obj,
     const jlong millis) {
   DCHECK(media_session_);
-  DCHECK_GT(millis, 0) << "Attempted to seek to a negative position";
+  DCHECK_GT(millis, -1) << "Attempted to seek to a negative position";
   media_session_->SeekTo(base::TimeDelta::FromMilliseconds(millis));
 }
 
