diff --git tb/components/password_manager/core/browser/password_form_manager.cc tb_diff/components/password_manager/core/browser/password_form_manager.cc
--- tb/components/password_manager/core/browser/password_form_manager.cc
+++ tb_diff/components/password_manager/core/browser/password_form_manager.cc
@@ -258,10 +258,12 @@ bool PasswordFormManager::IsBlacklisted() const {
 void PasswordFormManager::Save() {
   DCHECK_EQ(FormFetcher::State::NOT_WAITING, form_fetcher_->GetState());
   DCHECK(!client_->IsIncognito());
+#if defined(ENABLE_TRIPLE_BANANA)
   if (IsBlacklisted()) {
     password_save_manager_->Unblacklist(ConstructObservedFormDigest());
     newly_blacklisted_ = false;
   }
+#endif
 
   password_save_manager_->Save(observed_form_, *parsed_submitted_form_);
 
