diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/infobar/SavePasswordInfoBar.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/infobar/SavePasswordInfoBar.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/infobar/SavePasswordInfoBar.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/infobar/SavePasswordInfoBar.java
@@ -34,9 +34,22 @@ public class SavePasswordInfoBar extends ConfirmInfoBar {
     @Override
     public void createContent(InfoBarLayout layout) {
         super.createContent(layout);
+        if (!org.banana.cake.interfaces.BananaPasswordExtension.get().isSecureLoginEnabled()) {
+            org.banana.cake.interfaces.BananaPasswordExtension.get().setupSecureLoginSwitch(
+                    layout.addControlLayout().addSwitch(
+                            0, 0, null, R.id.infobar_extra_check, true));
+        }
         if (!TextUtils.isEmpty(mDetailsMessage)) {
             InfoBarControlLayout detailsMessageLayout = layout.addControlLayout();
             detailsMessageLayout.addDescription(mDetailsMessage);
         }
     }
+
+    @Override
+    protected void onButtonClicked(@ActionType int action) {
+        super.onButtonClicked(action);
+        if (action == ActionType.OK) {
+            org.banana.cake.interfaces.BananaPasswordExtension.get().setSecureLoginEnabled();
+        }
+    }
 }
