diff --git tb/components/autofill/content/renderer/autofill_agent.h tb_diff/components/autofill/content/renderer/autofill_agent.h
--- tb/components/autofill/content/renderer/autofill_agent.h
+++ tb_diff/components/autofill/content/renderer/autofill_agent.h
@@ -30,6 +30,10 @@
 #include "third_party/blink/public/web/web_input_element.h"
 #include "ui/accessibility/ax_mode.h"
 
+#if defined(ENABLE_TRIPLE_BANANA)
+#include "triple_banana/modules/public/interfaces.h"
+#endif
+
 namespace blink {
 class WebNode;
 class WebView;
@@ -357,6 +361,11 @@ class AutofillAgent : public content::RenderFrameObserver,
   blink::WebFormControlElement last_clicked_form_control_element_for_testing_;
   bool last_clicked_form_control_element_was_focused_for_testing_ = false;
 
+#if defined(ENABLE_TRIPLE_BANANA)
+  mojo::Remote<authentication::mojom::AuthenticationManager> manager_ =
+      AutoBind(authentication::mojom::AuthenticationManager);
+#endif
+
   FormTracker form_tracker_;
 
   // Whether or not we delay focus handling until scrolling occurs.
