diff --git tb/components/search_engines/template_url_prepopulate_data.cc tb_diff/components/search_engines/template_url_prepopulate_data.cc
--- tb/components/search_engines/template_url_prepopulate_data.cc
+++ tb_diff/components/search_engines/template_url_prepopulate_data.cc
@@ -454,6 +454,9 @@ const PrepopulatedEngine* const engines_KR[] = {
     &naver,
     &daum,
     &bing,
+#if defined(ENABLE_TRIPLE_BANANA)
+    &youtube,
+#endif
 };
 
 // Kuwait
