diff --git tb/chrome/browser/flag_descriptions.cc tb_diff/chrome/browser/flag_descriptions.cc
--- tb/chrome/browser/flag_descriptions.cc
+++ tb_diff/chrome/browser/flag_descriptions.cc
@@ -583,6 +583,11 @@ const char kEnableTLS13EarlyDataDescription[] =
     "during the handshake when resuming a connection to a compatible TLS 1.3 "
     "server.";
 
+const char kPostQuantumCECPQ2Name[] = "TLS Post-Quantum Confidentiality";
+const char kPostQuantumCECPQ2Description[] =
+    "This option enables a post-quantum (i.e. resistent to quantum computers) "
+    "key exchange algorithm in TLS (CECPQ2).";
+
 const char kWinrtGeolocationImplementationName[] =
     "WinRT Geolocation Implementation";
 const char kWinrtGeolocationImplementationDescription[] =
