diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
@@ -1658,7 +1658,11 @@ public class ToolbarManager implements ToolbarTabController, UrlFocusChangeListe
 
         mToolbar.updateButtonVisibility();
         mToolbar.updateBackButtonVisibility(currentTab != null && currentTab.canGoBack());
+        org.banana.cake.interfaces.BananaBottomToolbarController.get().updateBackButtonVisibility(
+                currentTab != null && currentTab.canGoBack());
         mToolbar.updateForwardButtonVisibility(currentTab != null && currentTab.canGoForward());
+        org.banana.cake.interfaces.BananaBottomToolbarController.get()
+                .updateForwardButtonVisibility(currentTab != null && currentTab.canGoForward());
         updateReloadState(tabCrashed);
         updateBookmarkButtonStatus();
         if (mToolbar.getMenuButtonWrapper() != null && !isBottomToolbarVisible()) {
@@ -1672,6 +1676,8 @@ public class ToolbarManager implements ToolbarTabController, UrlFocusChangeListe
         boolean editingAllowed = currentTab == null || mBookmarkBridge == null
                 || mBookmarkBridge.isEditBookmarksEnabled();
         mToolbar.updateBookmarkButton(isBookmarked, editingAllowed);
+        org.banana.cake.interfaces.BananaBottomToolbarController.get().updateBookmarkButtonStatus(
+                isBookmarked, editingAllowed);
     }
 
     private void updateReloadState(boolean tabCrashed) {
