diff --git tb/chrome/browser/password_manager/chrome_password_manager_client.cc tb_diff/chrome/browser/password_manager/chrome_password_manager_client.cc
--- tb/chrome/browser/password_manager/chrome_password_manager_client.cc
+++ tb_diff/chrome/browser/password_manager/chrome_password_manager_client.cc
@@ -343,7 +343,11 @@ bool ChromePasswordManagerClient::PromptUserToSaveOrUpdatePassword(
         std::move(form_to_save));
   }
 #else
+#if defined(ENABLE_TRIPLE_BANANA)
+  if (!update_password && form_to_save->IsBlacklisted())
+#else
   if (form_to_save->IsBlacklisted())
+#endif
     return false;
 
   if (update_password) {
