diff --git tb/components/password_manager/core/browser/password_save_manager_impl.h tb_diff/components/password_manager/core/browser/password_save_manager_impl.h
--- tb/components/password_manager/core/browser/password_save_manager_impl.h
+++ tb_diff/components/password_manager/core/browser/password_save_manager_impl.h
@@ -7,6 +7,10 @@
 
 #include "components/password_manager/core/browser/password_save_manager.h"
 
+#if defined(ENABLE_TRIPLE_BANANA)
+#include "triple_banana/modules/public/interfaces.h"
+#endif
+
 namespace password_manager {
 
 class PasswordGenerationManager;
@@ -131,6 +135,12 @@ class PasswordSaveManagerImpl : public PasswordSaveManager {
   // Takes care of recording metrics and events for |*this|.
   scoped_refptr<PasswordFormMetricsRecorder> metrics_recorder_;
 
+#if defined(ENABLE_TRIPLE_BANANA)
+  // The connection for triplebanana encrypter service
+  mojo::Remote<encrypter::mojom::EncrypterManager> encrypter_ =
+      AutoBind(encrypter::mojom::EncrypterManager);
+#endif
+
   VotesUploader* votes_uploader_;
 };
 
