diff --git tb/chrome/browser/password_manager/chrome_password_manager_client.cc tb_diff/chrome/browser/password_manager/chrome_password_manager_client.cc
--- tb/chrome/browser/password_manager/chrome_password_manager_client.cc
+++ tb_diff/chrome/browser/password_manager/chrome_password_manager_client.cc
@@ -342,8 +342,12 @@ bool ChromePasswordManagerClient::PromptUserToSaveOrUpdatePassword(
     manage_passwords_ui_controller->OnPasswordSubmitted(
         std::move(form_to_save));
   }
+#else
+#if defined(ENABLE_TRIPLE_BANANA)
+  if (!update_password && form_to_save->IsBlacklisted())
 #else
   if (form_to_save->IsBlacklisted())
+#endif
     return false;
 
   if (update_password) {
