diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
@@ -854,8 +854,8 @@ public class ToolbarManager implements ScrimObserver, ToolbarTabController, UrlF
                 && (!FeatureUtilities.isAdaptiveToolbarEnabled()
                         || mActivity.getResources().getConfiguration().orientation
                                 != Configuration.ORIENTATION_LANDSCAPE);
-        mBottomControlsCoordinator.setBottomControlsVisible(mIsBottomToolbarVisible);
-        mToolbar.onBottomToolbarVisibilityChanged(mIsBottomToolbarVisible);
+        mBottomControlsCoordinator.setBottomControlsVisible(true);
+        mToolbar.onBottomToolbarVisibilityChanged(false);
 
         Toast.setGlobalExtraYOffset(mActivity.getResources().getDimensionPixelSize(
                 FeatureUtilities.isLabeledBottomToolbarEnabled()
@@ -1359,8 +1359,8 @@ public class ToolbarManager implements ScrimObserver, ToolbarTabController, UrlF
                     && mIsBottomToolbarVisible) {
                 mIsBottomToolbarVisible = !mActivity.isInOverviewMode();
             }
-            mToolbar.onBottomToolbarVisibilityChanged(mIsBottomToolbarVisible);
-            mBottomControlsCoordinator.setBottomControlsVisible(mIsBottomToolbarVisible);
+            mToolbar.onBottomToolbarVisibilityChanged(false);
+            mBottomControlsCoordinator.setBottomControlsVisible(true);
             if (mAppMenuButtonHelper != null) {
                 mAppMenuButtonHelper.setMenuShowsFromBottom(isMenuFromBottom());
             }
