diff --git tb/chrome/android/BUILD.gn tb_diff/chrome/android/BUILD.gn
--- tb/chrome/android/BUILD.gn
+++ tb_diff/chrome/android/BUILD.gn
@@ -29,6 +29,7 @@ import("//third_party/icu/config.gni")
 import("//third_party/protobuf/proto_library.gni")
 import("//tools/resources/generate_resource_whitelist.gni")
 import("//tools/v8_context_snapshot/v8_context_snapshot.gni")
+import("//triple_banana/triple_banana.gni")
 import("//weblayer/variables.gni")
 import("channel.gni")
 import("java_sources.gni")
@@ -439,6 +440,7 @@ android_library("chrome_java") {
     "//third_party/gif_player:gif_player_java",
     "//third_party/google_android_play_core:com_google_android_play_core_java",
     "//third_party/jsr-305:jsr_305_javalib",
+    "//triple_banana/modules:mojom_java",
     "//ui/android:ui_java",
     "//ui/base/mojom:mojom_java",
     "//ui/gfx/geometry/mojom:mojom_java",
@@ -501,7 +503,7 @@ android_library("chrome_java") {
   sources = chrome_java_sources + [
               app_hooks_impl,
               _chrome_version_java_file,
-            ]
+            ] + triple_banana_hooks_sources
 
   # Include sources from feed_java_sources.gni.
   sources += feed_java_sources
