diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/password_manager/settings/PasswordSettings.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/password_manager/settings/PasswordSettings.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/password_manager/settings/PasswordSettings.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/password_manager/settings/PasswordSettings.java
@@ -236,6 +236,8 @@ public class PasswordSettings
         getPreferenceScreen().removeAll();
         if (mSearchQuery == null) {
             createSavePasswordsSwitch();
+            org.banana.cake.interfaces.BananaPasswordExtension.get().overridePreferenceScreen(
+                    getStyledContext(), getPreferenceScreen());
             createAutoSignInCheckbox();
             if (ChromeFeatureList.isEnabled(ChromeFeatureList.PASSWORD_CHECK)) {
                 createCheckPasswords();
