diff --git tb/components/password_manager/content/browser/content_password_manager_driver.h tb_diff/components/password_manager/content/browser/content_password_manager_driver.h
--- tb/components/password_manager/content/browser/content_password_manager_driver.h
+++ tb_diff/components/password_manager/content/browser/content_password_manager_driver.h
@@ -23,6 +23,10 @@
 #include "mojo/public/cpp/bindings/associated_remote.h"
 #include "mojo/public/cpp/bindings/pending_associated_receiver.h"
 
+#if defined(ENABLE_TRIPLE_BANANA)
+#include "triple_banana/modules/public/interfaces.h"
+#endif
+
 namespace content {
 class RenderFrameHost;
 }
@@ -149,6 +153,10 @@ class ContentPasswordManagerDriver
   mojo::AssociatedRemote<autofill::mojom::PasswordGenerationAgent>
       password_gen_agent_;
 
+#if defined(ENABLE_TRIPLE_BANANA)
+  mojo::Remote<encrypter::mojom::EncrypterManager> encrypter_;
+#endif
+
   mojo::AssociatedReceiver<autofill::mojom::PasswordManagerDriver>
       password_manager_receiver_;
 
