diff --git tb/chrome/browser/touch_to_fill/android/touch_to_fill_view_impl.cc tb_diff/chrome/browser/touch_to_fill/android/touch_to_fill_view_impl.cc
--- tb/chrome/browser/touch_to_fill/android/touch_to_fill_view_impl.cc
+++ tb_diff/chrome/browser/touch_to_fill/android/touch_to_fill_view_impl.cc
@@ -62,8 +62,7 @@ void TouchToFillViewImpl::Show(
     const GURL& url,
     IsOriginSecure is_origin_secure,
     base::span<const password_manager::UiCredential> credentials) {
-  auto java_object = GetOrCreateJavaObject();
-  if (!java_object)
+  if (!CreateJavaObject())
     return;
   // Serialize the |credentials| span into a Java array and instruct the bridge
   // to show it together with |url| to the user.
@@ -83,7 +82,7 @@ void TouchToFillViewImpl::Show(
   }
 
   Java_TouchToFillBridge_showCredentials(
-      env, java_object, ConvertUTF8ToJavaString(env, url.spec()),
+      env, java_object_internal_, ConvertUTF8ToJavaString(env, url.spec()),
       is_origin_secure.value(), credential_array);
 }
 
@@ -109,16 +108,18 @@ void TouchToFillViewImpl::OnDismiss(JNIEnv* env) {
   OnDismiss();
 }
 
-base::android::ScopedJavaGlobalRef<jobject>
-TouchToFillViewImpl::GetOrCreateJavaObject() {
-  if (java_object_internal_) {
-    return java_object_internal_;
-  }
+bool TouchToFillViewImpl::CreateJavaObject() {
   if (controller_->GetNativeView() == nullptr ||
       controller_->GetNativeView()->GetWindowAndroid() == nullptr) {
-    return nullptr;  // No window attached (yet or anymore).
+    return false;  // No window attached (yet or anymore).
+  }
+  if (java_object_internal_) {
+    Java_TouchToFillBridge_destroy(AttachCurrentThread(),
+                                   java_object_internal_);
+    java_object_internal_.Reset();
   }
-  return java_object_internal_ = Java_TouchToFillBridge_create(
+  java_object_internal_ = Java_TouchToFillBridge_create(
              AttachCurrentThread(), reinterpret_cast<intptr_t>(this),
              controller_->GetNativeView()->GetWindowAndroid()->GetJavaObject());
+  return !!java_object_internal_;
 }
