diff --git tb/chrome/browser/BUILD.gn tb_diff/chrome/browser/BUILD.gn
--- tb/chrome/browser/BUILD.gn
+++ tb_diff/chrome/browser/BUILD.gn
@@ -2143,6 +2143,7 @@ static_library("browser") {
     "//third_party/zlib:minizip",
     "//third_party/zlib/google:compression_utils",
     "//third_party/zlib/google:zip",
+    "//triple_banana/modules:native",
     "//ui/accessibility",
     "//ui/base",
     "//ui/base:ui_data_pack",
