diff --git tb/components/translate/core/browser/translate_manager.cc tb_diff/components/translate/core/browser/translate_manager.cc
--- tb/components/translate/core/browser/translate_manager.cc
+++ tb_diff/components/translate/core/browser/translate_manager.cc
@@ -308,7 +308,11 @@ bool TranslateManager::CanManuallyTranslate(bool menuLogging) {
 void TranslateManager::InitiateManualTranslation(bool auto_translate,
                                                  bool triggered_from_menu) {
   // If a translation has already been triggered, do nothing.
+#if !defined(ENABLE_TRIPLE_BANANA)
   if (language_state_.IsPageTranslated() ||
+#else
+  if (
+#endif  // #if !defined(ENABLE_TRIPLE_BANANA)
       language_state_.translation_pending())
     return;
 
