diff --git tb/chrome/browser/password_manager/chrome_password_manager_client.cc tb_diff/chrome/browser/password_manager/chrome_password_manager_client.cc
--- tb/chrome/browser/password_manager/chrome_password_manager_client.cc
+++ tb_diff/chrome/browser/password_manager/chrome_password_manager_client.cc
@@ -333,7 +333,11 @@ bool ChromePasswordManagerClient::PromptUserToSaveOrUpdatePassword(
     return false;
 
 #if defined(OS_ANDROID)
+#if defined(ENABLE_TRIPLE_BANANA)
+  if (!update_password && form_to_save->IsBlocklisted())
+#else
   if (form_to_save->IsBlocklisted())
+#endif
     return false;
 
   if (update_password) {
