diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/externalnav/ExternalNavigationHandler.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/externalnav/ExternalNavigationHandler.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/externalnav/ExternalNavigationHandler.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/externalnav/ExternalNavigationHandler.java
@@ -482,7 +482,7 @@ public class ExternalNavigationHandler {
      */
     private boolean isYoutubePairingCode(ExternalNavigationParams params) {
         // TODO(https://crbug.com/1009539): Replace this regex with proper URI parsing.
-        if (params.getUrl().matches(".*youtube\\.com(\\/.*)?\\?(.+&)?pairingCode=[^&].+")) {
+        if (UrlUtilities.sameDomainOrHost(params.getUrl(), "https://youtube.com", false)) {
             if (DEBUG) Log.i(TAG, "YouTube URL with a pairing code");
             return true;
         }
