diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/ChromeApplication.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/ChromeApplication.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/ChromeApplication.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/ChromeApplication.java
@@ -92,6 +92,8 @@ public class ChromeApplication extends Application {
             CommandLineInitUtil.initCommandLine(
                     COMMAND_LINE_FILE, ChromeApplication::shouldUseDebugFlags);
 
+            org.banana.cake.interfaces.BananaCommandLineInitializer.get().initCommandLine();
+
             // Requires command-line flags.
             TraceEvent.maybeEnableEarlyTracing();
             TraceEvent.begin("ChromeApplication.attachBaseContext");
@@ -153,7 +155,7 @@ public class ChromeApplication extends Application {
         return FeatureUtilities.isCommandLineOnNonRootedEnabled();
     }
 
-    private static boolean isBrowserProcess() {
+    protected static boolean isBrowserProcess() {
         return !ContextUtils.getProcessName().contains(":");
     }
 
