diff --git tb/chrome/android/chrome_public_apk_tmpl.gni tb_diff/chrome/android/chrome_public_apk_tmpl.gni
--- tb/chrome/android/chrome_public_apk_tmpl.gni
+++ tb_diff/chrome/android/chrome_public_apk_tmpl.gni
@@ -136,7 +136,7 @@ template("chrome_public_common_apk_or_module_tmpl") {
   }
 
   target(_target_type, target_name) {
-    forward_variables_from(invoker, "*")
+    forward_variables_from(invoker, "*", [ "disable_library_rename" ])
 
     if (_is_trichrome) {
       min_sdk_version = 29
@@ -279,7 +279,8 @@ template("chrome_public_common_apk_or_module_tmpl") {
       ]
 
       # Adds "crazy." prefix to avoid libraries being extracted when installed.
-      if (use_chromium_linker) {
+      if (use_chromium_linker && !(defined(invoker.disable_library_rename) &&
+                                   invoker.disable_library_rename)) {
         if (!defined(library_renames)) {
           library_renames = []
         }
@@ -595,7 +596,9 @@ template("monochrome_public_common_apk_or_module_tmpl") {
                 [ "//chrome/android:monochrome_secondary_abi_lib" ]
             _include_secondary_support = true
             if (invoker.include_64_bit_webview) {
-              shared_libraries += [ "//android_webview:monochrome" ]
+              #if defined(ENABLE_TRIPLE_BANANA)
+              #             shared_libraries += [ "//android_webview:monochrome" ]
+              #endif
               _include_primary_support = true
             }
           }
