diff --git tb/chrome/browser/ui/android/strings/android_chrome_strings.grd tb_diff/chrome/browser/ui/android/strings/android_chrome_strings.grd
--- tb/chrome/browser/ui/android/strings/android_chrome_strings.grd
+++ tb_diff/chrome/browser/ui/android/strings/android_chrome_strings.grd
@@ -2856,6 +2856,9 @@ To change this setting, <ph name="BEGIN_LINK">&lt;resetlink&gt;</ph>reset sync<p
       <message name="IDS_CUSTOM_TAB_AMP_PUBLISHER_URL" desc="Text shown in a Custom Tab toolbar for a web page that is hosted by the Google content delivery network but was originally published by someone else.">
         From <ph name="PUBLISHER_ORIGIN">&lt;pub&gt;%1$s&lt;/pub&gt;<ex>example.com</ex></ph> – <ph name="BEGIN_DEEMPHASIZED">&lt;bg&gt;</ph>delivered by Google<ph name="END_DEEMPHASIZED">&lt;/bg&gt;</ph>
       </message>
+      <message name="IDS_TOOLBAR_EDIT" desc="Edit activity for bottom toolbar">
+        Toolbar Edit
+      </message>
 
       <!-- Main menu items -->
       <message name="IDS_MENU_UPDATE" desc="Menu item for updating chrome. [CHAR-LIMIT=24]">
