diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/settings/password/SavePasswordsPreferences.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/settings/password/SavePasswordsPreferences.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/settings/password/SavePasswordsPreferences.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/settings/password/SavePasswordsPreferences.java
@@ -213,6 +213,8 @@ public class SavePasswordsPreferences
         getPreferenceScreen().removeAll();
         if (mSearchQuery == null) {
             createSavePasswordsSwitch();
+            org.banana.cake.interfaces.BananaPasswordExtension.get().overridePreferenceScreen(
+                    getStyledContext(), getPreferenceScreen());
             createAutoSignInCheckbox();
         }
         PasswordManagerHandlerProvider.getInstance()
