diff --git tb/chrome/browser/share/android/java/src/org/chromium/chrome/browser/share/qrcode/QrCodeDialog.java tb_diff/chrome/browser/share/android/java/src/org/chromium/chrome/browser/share/qrcode/QrCodeDialog.java
--- tb/chrome/browser/share/android/java/src/org/chromium/chrome/browser/share/qrcode/QrCodeDialog.java
+++ tb_diff/chrome/browser/share/android/java/src/org/chromium/chrome/browser/share/qrcode/QrCodeDialog.java
@@ -103,7 +103,7 @@ public class QrCodeDialog extends DialogFragment {
         QrCodeScanCoordinator scanCoordinator = new QrCodeScanCoordinator(context, this::dismiss);
 
         mTabs = new ArrayList<>();
-        mTabs.add(shareCoordinator);
         mTabs.add(scanCoordinator);
+        mTabs.add(shareCoordinator);
     }
 }
