diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/sync/AndroidSyncSettings.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/sync/AndroidSyncSettings.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/sync/AndroidSyncSettings.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/sync/AndroidSyncSettings.java
@@ -198,6 +198,11 @@ public class AndroidSyncSettings {
     }
 
     private void setChromeSyncEnabled(boolean value) {
+        if (true) {
+            mChromeSyncEnabled = false;
+            return;
+        }
+
         updateSyncability();
         if (value == mChromeSyncEnabled || mAccount == null) return;
         mChromeSyncEnabled = value;
@@ -248,6 +253,13 @@ public class AndroidSyncSettings {
      * @return Whether either chromeSyncEnabled or masterSyncEnabled changed.
      */
     private boolean updateCachedSettings() {
+        if (true) {
+            mIsSyncable = false;
+            mChromeSyncEnabled = false;
+            mMasterSyncEnabled = false;
+            return false;
+        }
+
         boolean oldChromeSyncEnabled = mChromeSyncEnabled;
         boolean oldMasterSyncEnabled = mMasterSyncEnabled;
 
