diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
@@ -45,7 +45,9 @@ public class PrivacySettings
     private static final String PREF_SECURE_DNS = "secure_dns";
     private static final String PREF_USAGE_STATS = "usage_stats_reporting";
     private static final String PREF_DO_NOT_TRACK = "do_not_track";
-    private static final String PREF_SYNC_AND_SERVICES_LINK = "sync_and_services_link";
+// #if !defined(ENABLE_TRIPLE_BANANA)
+//  private static final String PREF_SYNC_AND_SERVICES_LINK = "sync_and_services_link";
+// #endif
 
     private ManagedPreferenceDelegate mManagedPreferenceDelegate;
 
@@ -73,15 +75,17 @@ public class PrivacySettings
         Preference secureDnsPref = findPreference(PREF_SECURE_DNS);
         secureDnsPref.setVisible(privacyPrefManager.isDnsOverHttpsUiEnabled());
 
-        Preference syncAndServicesLink = findPreference(PREF_SYNC_AND_SERVICES_LINK);
-        NoUnderlineClickableSpan linkSpan = new NoUnderlineClickableSpan(getResources(), view -> {
-            SettingsLauncher settingsLauncher = new SettingsLauncherImpl();
-            settingsLauncher.launchSettingsActivity(getActivity(), SyncAndServicesSettings.class,
-                    SyncAndServicesSettings.createArguments(false));
-        });
-        syncAndServicesLink.setSummary(
-                SpanApplier.applySpans(getString(R.string.privacy_sync_and_services_link),
-                        new SpanApplier.SpanInfo("<link>", "</link>", linkSpan)));
+// #if !defined(ENABLE_TRIPLE_BANANA)
+//      Preference syncAndServicesLink = findPreference(PREF_SYNC_AND_SERVICES_LINK);
+//      NoUnderlineClickableSpan linkSpan = new NoUnderlineClickableSpan(getResources(), view -> {
+//          SettingsLauncher settingsLauncher = new SettingsLauncherImpl();
+//          settingsLauncher.launchSettingsActivity(getActivity(), SyncAndServicesSettings.class,
+//                  SyncAndServicesSettings.createArguments(false));
+//      });
+//      syncAndServicesLink.setSummary(
+//              SpanApplier.applySpans(getString(R.string.privacy_sync_and_services_link),
+//                      new SpanApplier.SpanInfo("<link>", "</link>", linkSpan)));
+// #endif
 
         updateSummaries();
     }
