diff --git tb/content/browser/media/session/media_session_impl.h tb_diff/content/browser/media/session/media_session_impl.h
--- tb/content/browser/media/session/media_session_impl.h
+++ tb_diff/content/browser/media/session/media_session_impl.h
@@ -399,6 +399,10 @@ class MediaSessionImpl : public MediaSession,
   // with this media session.
   void ForAllPlayers(base::RepeatingCallback<void(const PlayerIdentifier&)>);
 
+#if defined(ENABLE_TRIPLE_BANANA)
+  bool IsMediaRemoteEnabled() const;
+#endif
+
   // A set of actions supported by |routed_service_| and the current media
   // session.
   std::set<media_session::mojom::MediaSessionAction> actions_;
