diff --git tb/chrome/android/BUILD.gn tb_diff/chrome/android/BUILD.gn
--- tb/chrome/android/BUILD.gn
+++ tb_diff/chrome/android/BUILD.gn
@@ -28,6 +28,7 @@ import("//third_party/icu/config.gni")
 import("//third_party/protobuf/proto_library.gni")
 import("//tools/resources/generate_resource_allowlist.gni")
 import("//tools/v8_context_snapshot/v8_context_snapshot.gni")
+import("//triple_banana/triple_banana.gni")
 import("//weblayer/variables.gni")
 import("channel.gni")
 import("java_sources.gni")
@@ -522,6 +523,7 @@ android_library("chrome_java") {
     "//third_party/gif_player:gif_player_java",
     "//third_party/google_android_play_core:com_google_android_play_core_java",
     "//third_party/metrics_proto:metrics_proto_java",
+    "//triple_banana/modules:mojom_java",
     "//ui/android:ui_java",
     "//ui/base/mojom:mojom_java",
     "//ui/gfx/geometry/mojom:mojom_java",
@@ -565,6 +567,8 @@ android_library("chrome_java") {
   # From java_sources.gni.
   sources = chrome_java_sources + [ app_hooks_impl ]
 
+  sources += triple_banana_hooks_sources + triple_banana_jni_sources
+
   # Include sources from feed_java_sources.gni.
   sources += feed_java_sources
   srcjar_deps += feed_srcjar_deps
@@ -1505,7 +1509,10 @@ jinja_template_resources("chrome_public_apk_template_resources") {
     "java/res_template/xml/file_paths.xml",
     "java/res_template/xml/launchershortcuts.xml",
     "java/res_template/xml/searchable.xml",
-    "java/res_template/xml/syncadapter.xml",
+
+    #if defined(ENABLE_TRIPLE_BANANA)
+    #   "java/res_template/xml/syncadapter.xml",
+    #endif
   ]
   res_dir = "java/res_template"
   variables = [ "manifest_package=$chrome_public_manifest_package" ]
@@ -3404,6 +3411,8 @@ generate_jni("chrome_jni_headers") {
     "java/src/org/chromium/chrome/browser/webauth/Fido2Helper.java",
   ]
 
+  sources += triple_banana_jni_sources
+
   # Used for testing only, should not be shipped to end users.
   if (enable_offline_pages_harness) {
     sources += [ "java/src/org/chromium/chrome/browser/offlinepages/evaluation/OfflinePageEvaluationBridge.java" ]
