diff --git tb/components/password_manager/core/browser/form_parsing/form_parser.cc tb_diff/components/password_manager/core/browser/form_parsing/form_parser.cc
--- tb/components/password_manager/core/browser/form_parsing/form_parser.cc
+++ tb_diff/components/password_manager/core/browser/form_parsing/form_parser.cc
@@ -996,7 +996,11 @@ std::unique_ptr<PasswordForm> FormDataParser::Parse(const FormData& form_data,
                                     processed_fields, username_max);
       if (username_field_by_context &&
           !(mode == FormDataParser::Mode::kSaving &&
+#if defined(ENABLE_TRIPLE_BANANA)
+            GetFieldValue(*username_field_by_context).empty())) {
+#else
             username_field_by_context->value.empty())) {
+#endif
         significant_fields.username = username_field_by_context;
         if (method == UsernameDetectionMethod::kNoUsernameDetected ||
             method == UsernameDetectionMethod::kBaseHeuristic) {
