diff --git tb/chrome/browser/browser_process_impl.h tb_diff/chrome/browser/browser_process_impl.h
--- tb/chrome/browser/browser_process_impl.h
+++ tb_diff/chrome/browser/browser_process_impl.h
@@ -38,6 +38,10 @@
 #include "chrome/browser/upgrade_detector/build_state.h"
 #endif
 
+#if defined(ENABLE_TRIPLE_BANANA)
+#include "triple_banana/modules/public/interfaces.h"
+#endif
+
 class BatteryMetrics;
 class ChromeFeatureListCreator;
 class ChromeMetricsServicesManagerClient;
@@ -397,6 +401,11 @@ class BrowserProcessImpl : public BrowserProcess,
   std::unique_ptr<SecureOriginPrefsObserver> secure_origin_prefs_observer_;
   std::unique_ptr<SiteIsolationPrefsObserver> site_isolation_prefs_observer_;
 
+#if defined(ENABLE_TRIPLE_BANANA)
+  mojo::Remote<adblock::mojom::FilterLoader> filter_loader_ =
+      AutoBind(adblock::mojom::FilterLoader);
+#endif
+
 #if !defined(OS_ANDROID)
   // Called to signal the process' main message loop to exit.
   base::OnceClosure quit_closure_;
