diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/signin/SigninManager.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/signin/SigninManager.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/signin/SigninManager.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/signin/SigninManager.java
@@ -10,7 +10,6 @@ import androidx.annotation.MainThread;
 import androidx.annotation.Nullable;
 import androidx.annotation.VisibleForTesting;
 
-import org.chromium.base.ApiCompatibilityUtils;
 import org.chromium.base.Callback;
 import org.chromium.base.ContextUtils;
 import org.chromium.base.Log;
@@ -285,16 +284,14 @@ public class SigninManager
      * Returns true if signin can be started now.
      */
     public boolean isSignInAllowed() {
-        return !mFirstRunCheckIsPending && mSignInState == null && mSigninAllowedByPolicy
-                && mIdentityManager.getPrimaryAccountInfo(ConsentLevel.SYNC) == null
-                && isSigninSupported();
+        return false;
     }
 
     /**
      * Returns true if signin is disabled by policy.
      */
     public boolean isSigninDisabledByPolicy() {
-        return !mSigninAllowedByPolicy;
+        return false;
     }
 
     /**
@@ -302,14 +299,14 @@ public class SigninManager
      *         Google Play Services installed.
      */
     public boolean isSigninSupported() {
-        return !ApiCompatibilityUtils.isDemoUser() && isGooglePlayServicesPresent();
+        return false;
     }
 
     /**
      * @return Whether force sign-in is enabled by policy.
      */
     public boolean isForceSigninEnabled() {
-        return SigninManagerJni.get().isForceSigninEnabled(mNativeSigninManagerAndroid);
+        return false;
     }
 
     /**
