diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
@@ -854,8 +854,8 @@ public class ToolbarManager implements ScrimObserver, ToolbarTabController, UrlF
                 && (!FeatureUtilities.isAdaptiveToolbarEnabled()
                         || mActivity.getResources().getConfiguration().orientation
                                 != Configuration.ORIENTATION_LANDSCAPE);
-        mBottomControlsCoordinator.setBottomControlsVisible(mIsBottomToolbarVisible);
-        mToolbar.onBottomToolbarVisibilityChanged(mIsBottomToolbarVisible);
+        mBottomControlsCoordinator.setBottomControlsVisible(true);
+        mToolbar.onBottomToolbarVisibilityChanged(false);
 
         Toast.setGlobalExtraYOffset(mActivity.getResources().getDimensionPixelSize(
                 FeatureUtilities.isLabeledBottomToolbarEnabled()
@@ -1359,8 +1359,8 @@ public class ToolbarManager implements ScrimObserver, ToolbarTabController, UrlF
                     && mIsBottomToolbarVisible) {
                 mIsBottomToolbarVisible = !mActivity.isInOverviewMode();
             }
-            mToolbar.onBottomToolbarVisibilityChanged(mIsBottomToolbarVisible);
-            mBottomControlsCoordinator.setBottomControlsVisible(mIsBottomToolbarVisible);
+            mToolbar.onBottomToolbarVisibilityChanged(false);
+            mBottomControlsCoordinator.setBottomControlsVisible(true);
             if (mAppMenuButtonHelper != null) {
                 mAppMenuButtonHelper.setMenuShowsFromBottom(isMenuFromBottom());
             }
@@ -1867,7 +1867,11 @@ public class ToolbarManager implements ScrimObserver, ToolbarTabController, UrlF
 
         mToolbar.updateButtonVisibility();
         mToolbar.updateBackButtonVisibility(currentTab != null && currentTab.canGoBack());
+        org.banana.cake.interfaces.BananaBottomToolbarController.get().updateBackButtonVisibility(
+                currentTab != null && currentTab.canGoBack());
         mToolbar.updateForwardButtonVisibility(currentTab != null && currentTab.canGoForward());
+        org.banana.cake.interfaces.BananaBottomToolbarController.get()
+                .updateForwardButtonVisibility(currentTab != null && currentTab.canGoForward());
         updateReloadState(tabCrashed);
         updateBookmarkButtonStatus();
         if (mToolbar.getMenuButtonWrapper() != null && !isBottomToolbarVisible()) {
@@ -1883,6 +1887,8 @@ public class ToolbarManager implements ScrimObserver, ToolbarTabController, UrlF
         boolean editingAllowed = currentTab == null || mBookmarkBridge == null
                 || mBookmarkBridge.isEditBookmarksEnabled();
         mToolbar.updateBookmarkButton(isBookmarked, editingAllowed);
+        org.banana.cake.interfaces.BananaBottomToolbarController.get().updateBookmarkButtonStatus(
+                isBookmarked, editingAllowed);
     }
 
     private void updateReloadState(boolean tabCrashed) {
