diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
@@ -1246,7 +1246,11 @@ public class ToolbarManager implements UrlFocusChangeListener, ThemeColorObserve
 
         mToolbar.updateButtonVisibility();
         mToolbar.updateBackButtonVisibility(currentTab != null && currentTab.canGoBack());
+        org.banana.cake.interfaces.BananaBottomToolbarController.get().updateBackButtonVisibility(
+                currentTab != null && currentTab.canGoBack());
         mToolbar.updateForwardButtonVisibility(currentTab != null && currentTab.canGoForward());
+        org.banana.cake.interfaces.BananaBottomToolbarController.get()
+                .updateForwardButtonVisibility(currentTab != null && currentTab.canGoForward());
         updateReloadState(tabCrashed);
         updateBookmarkButtonStatus();
         if (mToolbar.getMenuButtonWrapper() != null && !isBottomToolbarVisible()) {
@@ -1262,6 +1266,8 @@ public class ToolbarManager implements UrlFocusChangeListener, ThemeColorObserve
         boolean editingAllowed =
                 currentTab == null || bridge == null || bridge.isEditBookmarksEnabled();
         mToolbar.updateBookmarkButton(isBookmarked, editingAllowed);
+        org.banana.cake.interfaces.BananaBottomToolbarController.get().updateBookmarkButtonStatus(
+                isBookmarked, editingAllowed);
     }
 
     private void updateReloadState(boolean tabCrashed) {
