diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedRootUiCoordinator.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedRootUiCoordinator.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedRootUiCoordinator.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedRootUiCoordinator.java
@@ -257,6 +257,10 @@ public class TabbedRootUiCoordinator extends RootUiCoordinator implements Native
     @Override
     protected void initializeToolbar() {
         super.initializeToolbar();
+        if (org.banana.cake.interfaces.BananaBottomToolbarController.get().isEnabled()) {
+            getToolbarManager().enableBottomControls();
+            return;
+        }
         if (!mActivity.isTablet()
                 && (TabUiFeatureUtilities.isTabGroupsAndroidEnabled()
                         || TabUiFeatureUtilities.isConditionalTabStripEnabled())) {
