diff --git tb/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java tb_diff/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
--- tb/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
+++ tb_diff/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
@@ -634,6 +634,11 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
                 OfflinePageUtils.showOfflineSnackbarIfNecessary(tab);
             }
 
+            @Override
+            public void onUrlUpdated(Tab tab) {
+                org.banana.cake.interfaces.BananaTabManager.get().notifyUrlUpdated(tab);
+            }
+
             @Override
             public void onCrash(Tab tab) {
                 postDeferredStartupIfNeeded();
@@ -902,6 +907,13 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
         getManualFillingComponent().onResume();
     }
 
+    public PictureInPictureController getPipController() {
+        if (mPictureInPictureController == null) {
+            mPictureInPictureController = new PictureInPictureController();
+        }
+        return mPictureInPictureController;
+    }
+
     @Override
     protected void onUserLeaveHint() {
         super.onUserLeaveHint();
@@ -2008,9 +2020,13 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
                     currentTab.getWebContents().getNavigationController().getUseDesktopUserAgent();
             currentTab.getWebContents().getNavigationController().setUseDesktopUserAgent(
                     !usingDesktopUserAgent, reloadOnChange);
+            org.banana.cake.interfaces.BananaBottomToolbarController.get()
+                    .updateDesktopViewButtonState(!usingDesktopUserAgent);
             RecordUserAction.record("MobileMenuRequestDesktopSite");
         } else if (id == R.id.reader_mode_prefs_id) {
             DomDistillerUIUtils.openSettings(currentTab.getWebContents());
+        } else if (id == R.id.banana_extension) {
+            org.banana.cake.interfaces.BananaExtensionSettings.get().open(this);
         } else {
             return false;
         }
