diff --git tb/content/public/android/java/src/org/chromium/content/browser/MediaSessionImpl.java tb_diff/content/public/android/java/src/org/chromium/content/browser/MediaSessionImpl.java
--- tb/content/public/android/java/src/org/chromium/content/browser/MediaSessionImpl.java
+++ tb_diff/content/public/android/java/src/org/chromium/content/browser/MediaSessionImpl.java
@@ -69,7 +69,7 @@ public class MediaSessionImpl extends MediaSession {
 
     @Override
     public void seek(long millis) {
-        assert millis == 0 : "Attempted to seek by an unspecified number of milliseconds";
+        assert millis != 0 : "Attempted to seek by an unspecified number of milliseconds";
         MediaSessionImplJni.get().seek(mNativeMediaSessionAndroid, MediaSessionImpl.this, millis);
     }
 
