diff --git tb/components/search_engines/template_url_prepopulate_data.cc tb_diff/components/search_engines/template_url_prepopulate_data.cc
--- tb/components/search_engines/template_url_prepopulate_data.cc
+++ tb_diff/components/search_engines/template_url_prepopulate_data.cc
@@ -506,11 +506,18 @@ const PrepopulatedEngine* const engines_KE[] = {
 
 // South Korea
 const PrepopulatedEngine* const engines_KR[] = {
+#if defined(ENABLE_TRIPLE_BANANA)
+// clang-format off
+#endif
     &google,
     &naver,
     &bing,
     &daum,
     &yahoo_jp,
+#if defined(ENABLE_TRIPLE_BANANA)
+    &youtube,
+// clang-format on
+#endif
 };
 
 // Kuwait
@@ -938,11 +945,18 @@ const PrepopulatedEngine* const engines_UA[] = {
 
 // United States
 const PrepopulatedEngine* const engines_US[] = {
+#if defined(ENABLE_TRIPLE_BANANA)
+// clang-format off
+#endif
     &google,
     &bing,
     &yahoo,
     &duckduckgo,
     &ecosia,
+#if defined(ENABLE_TRIPLE_BANANA)
+    &youtube,
+// clang-format on
+#endif
 };
 
 // Uruguay
