diff --git tb/gpu/config/gpu_util.cc tb_diff/gpu/config/gpu_util.cc
--- tb/gpu/config/gpu_util.cc
+++ tb_diff/gpu/config/gpu_util.cc
@@ -45,6 +45,9 @@
 #include "ui/gl/gl_switches.h"
 
 #if defined(OS_ANDROID)
+#if defined(ENABLE_TRIPLE_BANANA) && defined(ARCH_CPU_ARM64)
+#include "base/android/android_image_reader_compat.h"
+#endif
 #include "base/no_destructor.h"
 #include "base/synchronization/lock.h"
 #include "ui/gfx/android/android_surface_control_compat.h"
@@ -111,6 +114,11 @@ GpuFeatureStatus GetAndroidSurfaceControlFeatureStatus(
 #if !defined(OS_ANDROID)
   return kGpuFeatureStatusDisabled;
 #else
+#if defined(ENABLE_TRIPLE_BANANA) && defined(ARCH_CPU_ARM64)
+  if (blocklisted_features.count(GPU_FEATURE_TYPE_ANDROID_SURFACE_CONTROL))
+    return kGpuFeatureStatusBlocklisted;
+#endif
+
   if (!gpu_preferences.enable_android_surface_control)
     return kGpuFeatureStatusDisabled;
 
@@ -327,6 +335,13 @@ void AdjustGpuFeatureStatusToWorkarounds(GpuFeatureInfo* gpu_feature_info) {
     gpu_feature_info->status_values[GPU_FEATURE_TYPE_ACCELERATED_WEBGL2] =
         kGpuFeatureStatusBlocklisted;
   }
+
+#if defined(ENABLE_TRIPLE_BANANA) && defined(ARCH_CPU_ARM64)
+  if (!base::android::AndroidImageReader::GetInstance().IsSupported()) {
+    gpu_feature_info->status_values[GPU_FEATURE_TYPE_ANDROID_SURFACE_CONTROL] =
+        kGpuFeatureStatusBlocklisted;
+  }
+#endif
 }
 
 // Estimates roughly user total disk space by counting in the drives where
