diff --git tb/mojo/public/tools/bindings/generators/java_templates/interface_definition.tmpl tb_diff/mojo/public/tools/bindings/generators/java_templates/interface_definition.tmpl
--- tb/mojo/public/tools/bindings/generators/java_templates/interface_definition.tmpl
+++ tb_diff/mojo/public/tools/bindings/generators/java_templates/interface_definition.tmpl
@@ -37,17 +37,20 @@ interface {{method|interface_response_name}} extends org.chromium.mojo.bindings.
 {%- endmacro -%}
 
 {%- macro flags_for_method(method, is_request) -%}
-{{flags(method.response_parameters != None, is_request)}}
+{{flags(method.response_parameters != None, is_request, method.sync)}}
 {%- endmacro -%}
 
-{%- macro flags(use_response_flag, is_request) -%}
+{%- macro flags(use_response_flag, is_request, is_sync) -%}
 {%-  if not use_response_flag -%}
 org.chromium.mojo.bindings.MessageHeader.NO_FLAG
 {%-   elif is_request: -%}
 org.chromium.mojo.bindings.MessageHeader.MESSAGE_EXPECTS_RESPONSE_FLAG
 {%-   else -%}
 org.chromium.mojo.bindings.MessageHeader.MESSAGE_IS_RESPONSE_FLAG
-{%-   endif -%}
+{%-   if is_sync -%}
+ | org.chromium.mojo.bindings.MessageHeader.MESSAGE_IS_SYNC_FLAG
+{%-   endif  -%}
+{%-  endif -%}
 {%- endmacro -%}
 
 {%- macro manager_class(interface, fully_qualified=False) -%}
@@ -91,7 +94,11 @@ try {
     org.chromium.mojo.bindings.ServiceMessage messageWithHeader =
             message.asServiceMessage();
     org.chromium.mojo.bindings.MessageHeader header = messageWithHeader.getHeader();
-    if (!header.validateHeader({{flags(with_response, True)}})) {
+    int flags = {{flags(with_response, True)}};
+    if (header.hasFlag(org.chromium.mojo.bindings.MessageHeader.MESSAGE_IS_SYNC_FLAG)) {
+        flags = flags | org.chromium.mojo.bindings.MessageHeader.MESSAGE_IS_SYNC_FLAG;
+    }
+    if (!header.validateHeader(flags)) {
         return false;
     }
     switch(header.getType()) {
