diff --git tb/media/capture/video/android/java/src/org/chromium/media/VideoCaptureCamera2.java tb_diff/media/capture/video/android/java/src/org/chromium/media/VideoCaptureCamera2.java
--- tb/media/capture/video/android/java/src/org/chromium/media/VideoCaptureCamera2.java
+++ tb_diff/media/capture/video/android/java/src/org/chromium/media/VideoCaptureCamera2.java
@@ -1433,6 +1433,7 @@ public class VideoCaptureCamera2 extends VideoCapture {
 
         final int[] capabilities =
                 cameraCharacteristics.get(CameraCharacteristics.REQUEST_AVAILABLE_CAPABILITIES);
+        if (capabilities == null) return null;
         // Per-format frame rate via getOutputMinFrameDuration() is only available if the
         // property REQUEST_AVAILABLE_CAPABILITIES_MANUAL_SENSOR is set.
         boolean minFrameDurationAvailable = false;
