diff --git tb/net/socket/ssl_client_socket_impl.cc tb_diff/net/socket/ssl_client_socket_impl.cc
--- tb/net/socket/ssl_client_socket_impl.cc
+++ tb_diff/net/socket/ssl_client_socket_impl.cc
@@ -761,9 +761,22 @@ int SSLClientSocketImpl::Init() {
   DCHECK(!ssl_);
 
   SSLContext* context = SSLContext::GetInstance();
+#if defined(ENABLE_TRIPLE_BANANA)
+  if (!base::FeatureList::IsEnabled(features::kPostQuantumCECPQ2)) {
+    if (ssl_config_.secure_dns_mode == SecureDnsMode::kSecure) {
+      const int kCurves[] = {NID_CECPQ2, NID_X25519, NID_X9_62_prime256v1,
+                             NID_secp384r1};
+      SSL_CTX_set1_curves(context->ssl_ctx(), kCurves, base::size(kCurves));
+    } else {
+      SSL_CTX_set1_curves(context->ssl_ctx(), nullptr, 0);
+    }
+  }
+#endif
+
   crypto::OpenSSLErrStackTracer err_tracer(FROM_HERE);
 
   ssl_.reset(SSL_new(context->ssl_ctx()));
+
   if (!ssl_ || !context->SetClientSocketForSSL(ssl_.get(), this))
     return ERR_UNEXPECTED;
 
