diff --git tb/third_party/blink/renderer/core/frame/web_local_frame_impl.cc tb_diff/third_party/blink/renderer/core/frame/web_local_frame_impl.cc
--- tb/third_party/blink/renderer/core/frame/web_local_frame_impl.cc
+++ tb_diff/third_party/blink/renderer/core/frame/web_local_frame_impl.cc
@@ -258,6 +258,10 @@
 #include "third_party/blink/renderer/platform/weborigin/security_policy.h"
 #include "third_party/blink/renderer/platform/wtf/hash_map.h"
 
+#if defined(ENABLE_TRIPLE_BANANA)
+#include "third_party/blink/renderer/core/fullscreen/fullscreen.h"
+#endif
+
 namespace blink {
 
 static int g_frame_count = 0;
@@ -2383,6 +2387,17 @@ void WebLocalFrameImpl::SendOrientationChangeEvent() {
   if (!GetFrame() || !GetFrame()->DomWindow())
     return;
 
+#if defined(ENABLE_TRIPLE_BANANA)
+  if (RuntimeEnabledFeatures::MediaRemoteEnabled() &&
+      GetFrame()->GetDocument()) {
+    Element* element =
+        Fullscreen::FullscreenElementFrom(*GetFrame()->GetDocument());
+    if (IsA<HTMLVideoElement>(element) &&
+        Fullscreen::IsFullscreenElement(*element))
+      return;
+  }
+#endif
+
   // Screen Orientation API
   CoreInitializer::GetInstance().NotifyOrientationChanged(*GetFrame());
 
