diff --git tb/third_party/blink/renderer/core/html/media/html_video_element.cc tb_diff/third_party/blink/renderer/core/html/media/html_video_element.cc
--- tb/third_party/blink/renderer/core/html/media/html_video_element.cc
+++ tb_diff/third_party/blink/renderer/core/html/media/html_video_element.cc
@@ -64,6 +64,11 @@
 #include "third_party/blink/renderer/platform/runtime_enabled_features.h"
 #include "third_party/blink/renderer/platform/web_test_support.h"
 
+#if defined(ENABLE_TRIPLE_BANANA)
+#include "third_party/blink/renderer/core/fullscreen/fullscreen_media_controller.h"
+#include "triple_banana/modules/public/interfaces.h"
+#endif
+
 namespace blink {
 
 namespace {
@@ -271,6 +276,12 @@ void HTMLVideoElement::UpdatePictureInPictureAvailability() {
 void HTMLVideoElement::OnBecamePersistentVideo(bool value) {
   is_auto_picture_in_picture_ = value;
 
+#if defined(ENABLE_TRIPLE_BANANA)
+  if (RuntimeEnabledFeatures::MediaRemoteEnabled()) {
+    FullscreenMediaController::Get().DispatchChangedPIPMode(value);
+  }
+#endif
+
   if (value) {
     // Record the type of video. If it is already fullscreen, it is a video with
     // native controls, otherwise it is assumed to be with custom controls.
