diff --git tb/third_party/blink/renderer/core/html/BUILD.gn tb_diff/third_party/blink/renderer/core/html/BUILD.gn
--- tb/third_party/blink/renderer/core/html/BUILD.gn
+++ tb_diff/third_party/blink/renderer/core/html/BUILD.gn
@@ -599,6 +599,7 @@ blink_core_sources("html") {
     "//skia:skcms",
     "//third_party/blink/public/strings",
     "//third_party/blink/renderer/core/html/parser:parser",
+    "//triple_banana/modules:native",
   ]
 }
 
