diff --git tb/third_party/blink/renderer/core/fullscreen/fullscreen.h tb_diff/third_party/blink/renderer/core/fullscreen/fullscreen.h
--- tb/third_party/blink/renderer/core/fullscreen/fullscreen.h
+++ tb_diff/third_party/blink/renderer/core/fullscreen/fullscreen.h
@@ -85,6 +85,13 @@ class CORE_EXPORT Fullscreen final : public GarbageCollected<Fullscreen>,
       RequestType,
       ScriptState* state = nullptr,
       ExceptionState* exception_state = nullptr);
+#if defined(ENABLE_TRIPLE_BANANA)
+  static ScriptPromise RequestFullscreen(const FullscreenOptions*,
+                                         Element&,
+                                         RequestType,
+                                         ScriptState* state,
+                                         ExceptionState* exception_state);
+#endif
 
   static void FullyExitFullscreen(Document&, bool ua_originated = false);
   static ScriptPromise ExitFullscreen(Document&,
diff --git tb/third_party/blink/renderer/core/fullscreen/fullscreen_media_controller.cc tb_diff/third_party/blink/renderer/core/fullscreen/fullscreen_media_controller.cc
new file mode 100644
