diff --git tb/third_party/blink/renderer/core/fullscreen/fullscreen.cc tb_diff/third_party/blink/renderer/core/fullscreen/fullscreen.cc
--- tb/third_party/blink/renderer/core/fullscreen/fullscreen.cc
+++ tb_diff/third_party/blink/renderer/core/fullscreen/fullscreen.cc
@@ -59,6 +59,11 @@
 #include "third_party/blink/renderer/platform/heap/heap.h"
 #include "third_party/blink/renderer/platform/instrumentation/use_counter.h"
 
+#if defined(ENABLE_TRIPLE_BANANA)
+#include "third_party/blink/renderer/core/fullscreen/fullscreen_media_controller.h"
+#include "third_party/blink/renderer/core/html/media/html_video_element.h"
+#endif
+
 namespace blink {
 
 namespace {
@@ -528,7 +533,16 @@ Element* Fullscreen::FullscreenElementFrom(Document& document) {
       return element;
   }
 
+#if defined(ENABLE_TRIPLE_BANANA)
+  if (RuntimeEnabledFeatures::MediaRemoteEnabled()) {
+    Element* element = FullscreenMediaController::Get().GetVideoElement();
+    if (element && HasFullscreenFlag(*element))
+      return element;
+  }
   return nullptr;
+#else
+  return nullptr;
+#endif
 }
 
 // https://fullscreen.spec.whatwg.org/#fullscreen-element
@@ -583,6 +597,32 @@ ScriptPromise Fullscreen::RequestFullscreen(Element& pending,
                                             FullscreenRequestType request_type,
                                             ScriptState* script_state,
                                             ExceptionState* exception_state) {
+#if defined(ENABLE_TRIPLE_BANANA)
+  if (RuntimeEnabledFeatures::MediaRemoteEnabled()) {
+    auto* query_element = IsA<HTMLVideoElement>(pending)
+                              ? &pending
+                              : pending.QuerySelector("video");
+    if (query_element) {
+      auto* video_element = DynamicTo<HTMLVideoElement>(query_element);
+      video_element->blur();
+      if (IsFullscreenElement(*video_element) ||
+          video_element->videoVisibleSize().IsEmpty()) {
+        return ScriptPromise();
+      }
+      return RequestFullscreen(options, *video_element, request_type,
+                               script_state, exception_state);
+    }
+  }
+  return RequestFullscreen(options, pending, request_type, script_state,
+                           exception_state);
+}
+
+ScriptPromise Fullscreen::RequestFullscreen(const FullscreenOptions* options,
+                                            Element& pending,
+                                            FullscreenRequestType request_type,
+                                            ScriptState* script_state,
+                                            ExceptionState* exception_state) {
+#endif
   RequestFullscreenScope scope;
 
   // 1. Let |pending| be the context object.
@@ -654,6 +694,14 @@ ScriptPromise Fullscreen::RequestFullscreen(Element& pending,
     LocalFrame& frame = *window.GetFrame();
     frame.GetChromeClient().EnterFullscreen(frame, options, request_type);
 
+#if defined(ENABLE_TRIPLE_BANANA)
+    if (RuntimeEnabledFeatures::MediaRemoteEnabled() &&
+        IsA<HTMLVideoElement>(pending)) {
+      FullscreenMediaController::Get().DispatchEnteredVideoFullscreen(
+          DynamicTo<HTMLVideoElement>(pending));
+    }
+#endif
+
     // After the first fullscreen request, the user activation should be
     // consumed, and the following fullscreen requests should receive an error.
     if (!for_cross_process_descendant)
@@ -842,6 +890,17 @@ ScriptPromise Fullscreen::ExitFullscreen(Document& doc,
     return ScriptPromise();
   }
 
+#if defined(ENABLE_TRIPLE_BANANA)
+  if (!ua_originated && RuntimeEnabledFeatures::MediaRemoteEnabled() &&
+      IsA<HTMLVideoElement>(FullscreenElementFrom(doc))) {
+    if (exception_state) {
+      exception_state->ThrowDOMException(DOMExceptionCode::kNotAllowedError,
+                                         "Unable to exit fullscreen");
+    }
+    return ScriptPromise();
+  }
+#endif
+
   if (script_state)
     resolver = MakeGarbageCollected<ScriptPromiseResolver>(script_state);
 
@@ -922,6 +981,12 @@ void Fullscreen::DidExitFullscreen(Document& document) {
     for (const Member<PendingExit>& exit : exits)
       ContinueExitFullscreen(&document, exit, true /* resize */);
   }
+
+#if defined(ENABLE_TRIPLE_BANANA)
+  if (RuntimeEnabledFeatures::MediaRemoteEnabled()) {
+    FullscreenMediaController::Get().DispatchExitedVideoFullscreen();
+  }
+#endif
 }
 
 void Fullscreen::ContinueExitFullscreen(Document* doc,
