diff --git tb/third_party/blink/renderer/core/BUILD.gn tb_diff/third_party/blink/renderer/core/BUILD.gn
--- tb/third_party/blink/renderer/core/BUILD.gn
+++ tb_diff/third_party/blink/renderer/core/BUILD.gn
@@ -290,6 +290,7 @@ component("core") {
     "//third_party/blink/renderer/core/xml:xpath_generated",
     "//third_party/blink/renderer/platform",
     "//third_party/blink/renderer/platform/wtf",
+    "//triple_banana/modules:native",
     "//ui/accessibility:ax_base",
     "//ui/base/cursor:cursor_base",
     "//ui/base/cursor/mojom:cursor_type_blink",
