diff --git tb/third_party/blink/public/mojom/BUILD.gn tb_diff/third_party/blink/public/mojom/BUILD.gn
--- tb/third_party/blink/public/mojom/BUILD.gn
+++ tb_diff/third_party/blink/public/mojom/BUILD.gn
@@ -246,6 +246,8 @@ mojom("mojom_platform") {
     sources += [ "serial/serial.mojom" ]
   }
 
+  public_deps += [ "//triple_banana/modules:mojom" ]
+
   if (is_mac) {
     public_deps += [
       "//ui/base/mojom",
