diff --git tb/third_party/blink/public/mojom/BUILD.gn tb_diff/third_party/blink/public/mojom/BUILD.gn
--- tb/third_party/blink/public/mojom/BUILD.gn
+++ tb_diff/third_party/blink/public/mojom/BUILD.gn
@@ -214,6 +214,7 @@ mojom("mojom_platform") {
     "//third_party/blink/public/mojom/dom_storage",
     "//third_party/blink/public/mojom/frame",
     "//third_party/blink/public/mojom/usb",
+    "//triple_banana/modules:mojom",
     "//ui/base/cursor/mojom",
     "//ui/base/ime/mojom",
     "//ui/display/mojom",
