diff --git tb/third_party/blink/public/BUILD.gn tb_diff/third_party/blink/public/BUILD.gn
--- tb/third_party/blink/public/BUILD.gn
+++ tb_diff/third_party/blink/public/BUILD.gn
@@ -52,6 +52,7 @@ group("all_blink") {
     "//third_party/blink/renderer/platform:blink_platform_unittests",
     "//third_party/blink/renderer/platform/heap:blink_heap_unittests",
     "//third_party/blink/renderer/platform/wtf:wtf_unittests",
+    "//triple_banana/modules:native",
   ]
 }
 
