diff --git tb/weblayer/variables.gni tb_diff/weblayer/variables.gni
--- tb/weblayer/variables.gni
+++ tb_diff/weblayer/variables.gni
@@ -6,7 +6,7 @@ import("//build/config/android/channel.gni")
 
 declare_args() {
   # Include the //weblayer code in WebView implementation APKs.
-  webview_includes_weblayer = true
+  webview_includes_weblayer = false
 
   # Whether WebLayer will be included as a DFM.
   weblayer_in_split = android_channel != "stable"
