diff --git triple_banana/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java
index 2b0d5f9..3536f43 100644
--- triple_banana/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java
+++ triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java
@@ -210,6 +210,9 @@ public class SavePasswordsPreferences
         getPreferenceScreen().removeAll();
         if (mSearchQuery == null) {
             createSavePasswordsSwitch();
+            getPreferenceScreen().addPreference(
+                    org.triple.banana.hooks_api.ChromeHooks.getInstance()
+                            .createAuthenticationSwitch(getStyledContext()));
             createAutoSignInCheckbox();
         }
         PasswordManagerHandlerProvider.getInstance()
