diff --git triple_banana/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java
index 1142f3f36600..9da287a5f7bd 100644
--- triple_banana/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java
+++ triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java
@@ -209,6 +209,9 @@ public class SavePasswordsPreferences
         getPreferenceScreen().removeAll();
         if (mSearchQuery == null) {
             createSavePasswordsSwitch();
+            getPreferenceScreen().addPreference(
+                    org.triple.banana.hooks_api.ChromeHooks.getInstance()
+                            .createAuthenticationSwitch(getStyledContext()));
             createAutoLeakDetectionSwitch();
             createAutoSignInCheckbox();
         }
-- 
2.23.0

