diff --git triple_banana/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java
index 2b0d5f9..0d8eb43 100644
--- triple_banana/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java
+++ triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/preferences/password/SavePasswordsPreferences.java
@@ -210,6 +210,8 @@ public class SavePasswordsPreferences
         getPreferenceScreen().removeAll();
         if (mSearchQuery == null) {
             createSavePasswordsSwitch();
+            org.banana.cake.interfaces.BananaPasswordExtension.get().overridePreferenceScreen(
+                    getStyledContext(), getPreferenceScreen());
             createAutoSignInCheckbox();
         }
         PasswordManagerHandlerProvider.getInstance()
