diff --git triple_banana/components/subresource_filter/content/browser/verified_ruleset_dealer.cc triple_banana_change/components/subresource_filter/content/browser/verified_ruleset_dealer.cc
index 4ab0e55..7ec8e8b 100644
--- triple_banana/components/subresource_filter/content/browser/verified_ruleset_dealer.cc
+++ triple_banana_change/components/subresource_filter/content/browser/verified_ruleset_dealer.cc
@@ -39,6 +39,8 @@ base::File VerifiedRulesetDealer::OpenAndSetRulesetFile(
     SetRulesetFile(file.Duplicate());
     expected_checksum_ = expected_checksum;
   }
+  // TODO(zino): We might have to make a callback here to detect when the
+  // ruleset is updated.
   return file;
 }
 
