diff --git triple_banana/content/browser/renderer_host/render_process_host_impl.cc triple_banana_change/content/browser/renderer_host/render_process_host_impl.cc
index 8379cdf..7a28c48 100644
--- triple_banana/content/browser/renderer_host/render_process_host_impl.cc
+++ triple_banana_change/content/browser/renderer_host/render_process_host_impl.cc
@@ -2858,6 +2858,7 @@ void RenderProcessHostImpl::AppendRendererCommandLine(
       switches::kDeviceScaleFactor,
       base::NumberToString(display::win::GetDPIScale()));
 #endif
+  command_line->AppendSwitch(switches::kDisableMediaSuspend);
 
   AppendCompositorCommandLineFlags(command_line);
 
