diff --git triple_banana/chrome/android/java/src/org/chromium/chrome/browser/ChromeActivity.java triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/ChromeActivity.java
index 99cac3d..e57ebc3 100644
--- triple_banana/chrome/android/java/src/org/chromium/chrome/browser/ChromeActivity.java
+++ triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/ChromeActivity.java
@@ -699,6 +699,11 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
             }
 
             @Override
+            public void onUrlUpdated(Tab tab) {
+                org.banana.cake.interfaces.BananaTabManager.get().notifyUrlUpdated(tab);
+            }
+
+            @Override
             public void onCrash(Tab tab) {
                 postDeferredStartupIfNeeded();
             }
@@ -2259,6 +2264,8 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
             RecordUserAction.record("MobileMenuRequestDesktopSite");
         } else if (id == R.id.reader_mode_prefs_id) {
             DomDistillerUIUtils.openSettings(currentTab.getWebContents());
+        } else if (id == R.id.toolbar_edit) {
+            org.banana.cake.interfaces.BananaToolbarEditor.get().show(this);
         } else {
             return false;
         }
