diff --git triple_banana/chrome/android/java/src/org/chromium/chrome/browser/util/FeatureUtilities.java triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/util/FeatureUtilities.java
index ee8d2cf..ba286df 100644
--- triple_banana/chrome/android/java/src/org/chromium/chrome/browser/util/FeatureUtilities.java
+++ triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/util/FeatureUtilities.java
@@ -378,7 +378,7 @@ public class FeatureUtilities {
      */
     public static boolean isBottomToolbarEnabled() {
         // TODO(crbug.com/944228): TabGroupsAndroid and ChromeDuet are incompatible for now.
-        return isFlagEnabled(BOTTOM_TOOLBAR_ENABLED_KEY, false)
+        return isFlagEnabled(BOTTOM_TOOLBAR_ENABLED_KEY, true)
                 && !DeviceFormFactor.isNonMultiDisplayContextOnTablet(
                         ContextUtils.getApplicationContext())
                 && (isDuetTabStripIntegrationAndroidEnabled() || !isTabGroupsAndroidEnabled());
