diff --git triple_banana/components/password_manager/core/browser/form_saver_impl.h triple_banana_change/components/password_manager/core/browser/form_saver_impl.h
index 992e9938bc6b..5380ae0a1dbb 100644
--- triple_banana/components/password_manager/core/browser/form_saver_impl.h
+++ triple_banana_change/components/password_manager/core/browser/form_saver_impl.h
@@ -8,7 +8,10 @@
 #include <memory>
 
 #include "base/macros.h"
+#include "base/strings/utf_string_conversions.h"
 #include "components/password_manager/core/browser/form_saver.h"
+#include "triple_banana/modules/public/connector.h"
+#include "triple_banana/modules/public/coroutine.h"
 
 namespace password_manager {
 
@@ -47,6 +50,8 @@ class FormSaverImpl : public FormSaver {
   // Cached pointer to the PasswordStore.
   PasswordStore* const store_;
 
+  mojo::InterfacePtr<encrypter::mojom::EncrypterManager> encrypter_;
+
   DISALLOW_COPY_AND_ASSIGN(FormSaverImpl);
 };
 
diff --git triple_banana/components/password_manager/core/browser/form_saver_impl.h triple_banana_change/components/password_manager/core/browser/form_saver_impl.h
index 5380ae0a1dbb..82dbd4c35a99 100644
--- triple_banana/components/password_manager/core/browser/form_saver_impl.h
+++ triple_banana_change/components/password_manager/core/browser/form_saver_impl.h
@@ -8,10 +8,8 @@
 #include <memory>
 
 #include "base/macros.h"
-#include "base/strings/utf_string_conversions.h"
 #include "components/password_manager/core/browser/form_saver.h"
-#include "triple_banana/modules/public/connector.h"
-#include "triple_banana/modules/public/coroutine.h"
+#include "triple_banana/modules/public/mojom/encrypter.mojom.h"
 
 namespace password_manager {
 
-- 
2.23.0

