diff --git triple_banana/content/browser/renderer_host/render_process_host_impl.cc triple_banana_change/content/browser/renderer_host/render_process_host_impl.cc
index bb73b8a..1f5de94 100644
--- triple_banana/content/browser/renderer_host/render_process_host_impl.cc
+++ triple_banana_change/content/browser/renderer_host/render_process_host_impl.cc
@@ -2901,6 +2901,7 @@ void RenderProcessHostImpl::AppendRendererCommandLine(
       switches::kDeviceScaleFactor,
       base::NumberToString(display::win::GetDPIScale()));
 #endif
+  command_line->AppendSwitch(switches::kDisableMediaSuspend);
 
   AppendCompositorCommandLineFlags(command_line);
 
