diff --git triple_banana/components/password_manager/core/browser/form_saver_impl.h triple_banana_change/components/password_manager/core/browser/form_saver_impl.h
index 992e993..82dbd4c 100644
--- triple_banana/components/password_manager/core/browser/form_saver_impl.h
+++ triple_banana_change/components/password_manager/core/browser/form_saver_impl.h
@@ -9,6 +9,7 @@
 
 #include "base/macros.h"
 #include "components/password_manager/core/browser/form_saver.h"
+#include "triple_banana/modules/public/mojom/encrypter.mojom.h"
 
 namespace password_manager {
 
@@ -47,6 +48,8 @@ class FormSaverImpl : public FormSaver {
   // Cached pointer to the PasswordStore.
   PasswordStore* const store_;
 
+  mojo::InterfacePtr<encrypter::mojom::EncrypterManager> encrypter_;
+
   DISALLOW_COPY_AND_ASSIGN(FormSaverImpl);
 };
 
