diff --git triple_banana/content/browser/builtin_service_manifests.cc triple_banana_change/content/browser/builtin_service_manifests.cc
index 8f9164d..773fbc0 100644
--- triple_banana/content/browser/builtin_service_manifests.cc
+++ triple_banana_change/content/browser/builtin_service_manifests.cc
@@ -29,6 +29,8 @@
 #include "services/service_manager/public/cpp/manifest_builder.h"
 #include "services/tracing/manifest.h"
 
+#include "triple_banana/modules/public/manifest/manifest.h"
+
 namespace content {
 
 namespace {
@@ -75,6 +77,7 @@ const std::vector<service_manager::Manifest>& GetBuiltinServiceManifests() {
                         kOutOfProcessBuiltin),
           resource_coordinator::GetManifest(),
           tracing::GetManifest(),
+          triple_banana::GetManifest(),
       }};
   return *manifests;
 }
