diff --git triple_banana/content/public/app/content_browser_manifest.cc triple_banana_change/content/public/app/content_browser_manifest.cc
index 9002c33..0ad940c 100644
--- triple_banana/content/public/app/content_browser_manifest.cc
+++ triple_banana_change/content/public/app/content_browser_manifest.cc
@@ -149,6 +149,9 @@ const service_manager::Manifest& GetContentBrowserManifest() {
           .RequireCapability("device", "device:usb_test")
           .RequireCapability("device", "device:vibration")
           .RequireCapability("device", "device:wake_lock")
+#if defined(ENABLE_TRIPLE_BANANA)
+          .RequireCapability("triple_banana", "modules")
+#endif
           .RequireCapability("media_session", "media_session:app")
           .RequireCapability("video_capture", "capture")
           .RequireCapability("video_capture", "tests")
