diff --git triple_banana/chrome/android/BUILD.gn triple_banana_change/chrome/android/BUILD.gn
index f95b9db..19bcc5aa 100644
--- triple_banana/chrome/android/BUILD.gn
+++ triple_banana_change/chrome/android/BUILD.gn
@@ -25,6 +25,7 @@ import("//third_party/icu/config.gni")
 import("//third_party/protobuf/proto_library.gni")
 import("//tools/resources/generate_resource_whitelist.gni")
 import("//tools/v8_context_snapshot/v8_context_snapshot.gni")
+import("//triple_banana/triple_banana.gni")
 import("//ui/android/features.gni")
 import("channel.gni")
 import("java_sources.gni")
@@ -88,10 +89,7 @@ generate_ui_locale_resources("ui_locale_string_resources") {
 }
 
 android_resources("chrome_app_java_resources") {
-  resource_dirs = [
-    "java/res",
-    "//chrome/android/java/res_chromium",
-  ]
+  resource_dirs = [ "java/res" ] + triple_banana_resource
 
   if (enable_android_night_mode) {
     resource_dirs += [ "//chrome/android/java/res_night" ]
@@ -388,6 +386,7 @@ android_library("chrome_java") {
     "//third_party/gif_player:gif_player_java",
     "//third_party/google_android_play_core:com_google_android_play_core_java",
     "//third_party/jsr-305:jsr_305_javalib",
+    "//triple_banana/modules:mojom_java",
     "//ui/android:ui_java",
     "//ui/base/mojom:mojom_java",
     "//ui/gfx/geometry/mojom:mojom_java",
@@ -449,7 +448,7 @@ android_library("chrome_java") {
   java_files = chrome_java_sources + [
                  app_hooks_impl,
                  _chrome_version_java_file,
-               ]
+               ] + triple_banana_hooks_sources
 
   # Include sources from feed_java_sources.gni.
   java_files += feed_java_sources
@@ -1088,7 +1087,7 @@ if (enable_vr || enable_arcore) {
 
 # Overrides icon / name defined in chrome_app_java_resources.
 android_resources("chrome_public_apk_resources") {
-  resource_dirs = [ "java/res_chromium" ]
+  resource_dirs = triple_banana_resource
 
   # Dep needed to ensure override works properly.
   deps = [
