diff --git triple_banana/components/autofill/content/renderer/password_autofill_agent.cc triple_banana_change/components/autofill/content/renderer/password_autofill_agent.cc
index ba73837..90868d1 100644
--- triple_banana/components/autofill/content/renderer/password_autofill_agent.cc
+++ triple_banana_change/components/autofill/content/renderer/password_autofill_agent.cc
@@ -1741,6 +1741,9 @@ bool PasswordAutofillAgent::FillUserNameAndPassword(
     WebInputElement password_element,
     const PasswordFormFillData& fill_data,
     RendererSavePasswordProgressLogger* logger) {
+#if 1
+  return false;
+#elif 0
   if (logger)
     logger->LogMessage(Logger::STRING_FILL_USERNAME_AND_PASSWORD_METHOD);
 
@@ -1863,6 +1866,7 @@ bool PasswordAutofillAgent::FillUserNameAndPassword(
 
   LogFirstFillingResult(fill_data, FillingResult::kSuccess);
   return true;
+#endif
 }
 
 void PasswordAutofillAgent::LogPrefilledUsernameFillOutcome(
-- 
2.7.4

