diff --git triple_banana/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarCoordinator.java triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarCoordinator.java
index 3e6369a..70acbb7 100644
--- triple_banana/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarCoordinator.java
+++ triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarCoordinator.java
@@ -23,9 +23,8 @@ import org.chromium.chrome.browser.toolbar.TabCountProvider;
  * The root coordinator for the bottom toolbar. It has two sub-components: the browsing mode bottom
  * toolbar and the tab switcher mode bottom toolbar.
  */
-class BottomToolbarCoordinator {
-    /** The browsing mode bottom toolbar component */
-    private final BrowsingModeBottomToolbarCoordinator mBrowsingModeCoordinator;
+public class BottomToolbarCoordinator {
+    private final org.banana.cake.interfaces.BananaBottomToolbarController mBrowsingModeCoordinator;
 
     /** The tab switcher mode bottom toolbar component */
     private TabSwitcherBottomToolbarCoordinator mTabSwitcherModeCoordinator;
@@ -51,9 +50,9 @@ class BottomToolbarCoordinator {
             ThemeColorProvider themeColorProvider) {
         View root = stub.inflate();
 
-        mBrowsingModeCoordinator = new BrowsingModeBottomToolbarCoordinator(root, tabProvider,
-                homeButtonListener, searchAcceleratorListener, shareButtonListener,
-                tabSwitcherLongClickListener);
+        mBrowsingModeCoordinator =
+                org.banana.cake.interfaces.BananaBottomToolbarController.get().init(
+                        root, tabProvider);
 
         mTabSwitcherModeStub = root.findViewById(R.id.bottom_toolbar_tab_switcher_mode_stub);
 
