diff --git triple_banana/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarCoordinator.java triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarCoordinator.java
index 3e6369a..e146dc1d 100644
--- triple_banana/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarCoordinator.java
+++ triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarCoordinator.java
@@ -23,9 +23,8 @@ import org.chromium.chrome.browser.toolbar.TabCountProvider;
  * The root coordinator for the bottom toolbar. It has two sub-components: the browsing mode bottom
  * toolbar and the tab switcher mode bottom toolbar.
  */
-class BottomToolbarCoordinator {
-    /** The browsing mode bottom toolbar component */
-    private final BrowsingModeBottomToolbarCoordinator mBrowsingModeCoordinator;
+public class BottomToolbarCoordinator {
+    private final BottomToolbarCoordinatorDelegate mBrowsingModeCoordinator;
 
     /** The tab switcher mode bottom toolbar component */
     private TabSwitcherBottomToolbarCoordinator mTabSwitcherModeCoordinator;
@@ -36,6 +35,18 @@ class BottomToolbarCoordinator {
     /** A provider that notifies components when the theme color changes.*/
     private final ThemeColorProvider mThemeColorProvider;
 
+    public interface BottomToolbarCoordinatorDelegate {
+        void initializeWithNative(OnClickListener tabSwitcherListener,
+                AppMenuButtonHelper menuButtonHelper, OverviewModeBehavior overviewModeBehavior,
+                TabCountProvider tabCountProvider, ThemeColorProvider themeColorProvider,
+                IncognitoStateProvider incognitoStateProvider);
+        void showAppMenuUpdateBadge();
+        void removeAppMenuUpdateBadge();
+        boolean isShowingAppMenuUpdateBadge();
+        MenuButton getMenuButton();
+        void destroy();
+    }
+
     /**
      * Build the coordinator that manages the bottom toolbar.
      * @param stub The bottom toolbar {@link ViewStub} to inflate.
@@ -51,9 +62,9 @@ class BottomToolbarCoordinator {
             ThemeColorProvider themeColorProvider) {
         View root = stub.inflate();
 
-        mBrowsingModeCoordinator = new BrowsingModeBottomToolbarCoordinator(root, tabProvider,
-                homeButtonListener, searchAcceleratorListener, shareButtonListener,
-                tabSwitcherLongClickListener);
+        mBrowsingModeCoordinator =
+                org.triple.banana.hooks_api.ChromeHooks.getInstance()
+                        .createBottomToolbarCoordinatorDelegate(root, tabProvider);
 
         mTabSwitcherModeStub = root.findViewById(R.id.bottom_toolbar_tab_switcher_mode_stub);
 
