diff --git triple_banana/content/browser/BUILD.gn triple_banana_change/content/browser/BUILD.gn
index 4f44dae..9361cd2 100644
--- triple_banana/content/browser/BUILD.gn
+++ triple_banana_change/content/browser/BUILD.gn
@@ -213,6 +213,7 @@ jumbo_source_set("browser") {
     "//third_party/zlib",
     "//third_party/zlib/google:compression_utils",
     "//third_party/zlib/google:zip",
+    "//triple_banana/modules:manifest",
     "//ui/accessibility",
     "//ui/accessibility:ax_enums_mojo",
     "//ui/base",
