diff --git triple_banana/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java
index cbc536b..3344df1 100644
--- triple_banana/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java
+++ triple_banana_change/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java
@@ -1568,6 +1568,7 @@ public class ChromeTabbedActivity extends ChromeActivity implements ScreenshotMo
                 commandLine.appendSwitchesAndArguments(args);
             }
         }
+        org.triple.banana.hooks_api.ChromeHooks.getInstance().initCommandLine();
 
         supportRequestWindowFeature(Window.FEATURE_ACTION_MODE_OVERLAY);
 
