diff --git triple_banana/build/config/BUILD.gn triple_banana_change/build/config/BUILD.gn
index 8b0501f..ddbdeb6 100644
--- triple_banana/build/config/BUILD.gn
+++ triple_banana_change/build/config/BUILD.gn
@@ -24,6 +24,8 @@ if (is_android) {
   import("//build/config/android/abi.gni")
 }
 
+import("//triple_banana/build/config/triple_banana_build.gni")
+
 declare_args() {
   # When set (the default) enables C++ iterator debugging in debug builds.
   # Iterator debugging is always off in release builds (technically, this flag
@@ -124,6 +126,10 @@ config("feature_flags") {
     defines += [ "GOOGLE_CHROME_BUILD" ]
   }
 
+  if (enable_triple_banana) {
+    defines += triple_banana_build_flags
+  }
+
   # ==============================================
   #   PLEASE DO NOT ADD MORE THINGS TO THIS LIST
   # ==============================================
