diff --git triple_banana/components/subresource_filter/core/common/common_features.cc triple_banana_change/components/subresource_filter/core/common/common_features.cc
index 9a3b1a6..91061d7 100644
--- triple_banana/components/subresource_filter/core/common/common_features.cc
+++ triple_banana_change/components/subresource_filter/core/common/common_features.cc
@@ -6,6 +6,6 @@
 
 namespace subresource_filter {
 
-const base::Feature kAdTagging{"AdTagging", base::FEATURE_ENABLED_BY_DEFAULT};
+const base::Feature kAdTagging{"AdTagging", base::FEATURE_DISABLED_BY_DEFAULT};
 
 }  // namespace subresource_filter
