diff --git triple_banana/chrome/browser/chrome_content_browser_client.cc triple_banana_change/chrome/browser/chrome_content_browser_client.cc
index 030c3af..ebdf3dd 100644
--- triple_banana/chrome/browser/chrome_content_browser_client.cc
+++ triple_banana_change/chrome/browser/chrome_content_browser_client.cc
@@ -612,6 +612,8 @@
 #include "third_party/blink/public/mojom/page/spatial_navigation.mojom.h"
 #endif
 
+#include "triple_banana/modules/public/module_service.h"
+
 using base::FileDescriptor;
 using content::BrowserThread;
 using content::BrowserURLHandler;
@@ -3933,6 +3935,12 @@ void ChromeContentBrowserClient::RunServiceInstance(
   if (service)
     service_manager::Service::RunAsyncUntilTermination(std::move(service));
 #endif  // defined(OS_CHROMEOS)
+
+  if (service_name == "triple_banana") {
+    service_manager::Service::RunAsyncUntilTermination(
+        std::make_unique<triple_banana::ModuleService>(std::move(*receiver)));
+    return;
+  }
 }
 
 base::Optional<service_manager::Manifest>
